pkg/cluster/sync.go

```diff
@@ -393,6 +393,11 @@ func (c *Cluster) syncStatefulSet() error {
 		return fmt.Errorf("could not set cluster-wide PostgreSQL configuration options: %v", err)
 	}
 
+	// Apply Standby Cluster parameters
+	if err := c.checkAndSetStandbyClusterConfiguration(); err != nil {
+		return fmt.Errorf("could not set StandbyCluster configuration options: %v", err)
+	}
```

Comment:
Adding or removing the standby section must happen after the pod rotation. Otherwise you promote the standby and then immediately do a rolling update.